### PR TITLE
Console app

### PR DESCRIPTION
**Pull request by Alejandro-Bernal-M**
# basic UI of the app implemented on the console

image

## Create file app.rb with the logic
https://github.com/Alejandro-Bernal-M/school_library/blob/545570d46f7353c997ac48539cca85d45c6338b4/app.rb#L1-L172

## Calling the app class on the main.rb file
https://github.com/Alejandro-Bernal-M/school_library/blob/545570d46f7353c997ac48539cca85d45c6338b4/main.rb#L1-L8

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  end
+
+  def run
+    show_menu
+    if @number.to_i == 7
+      exit
+    else
+      options
+    end
+  end
```
Comment (franclobo) [inline app.rb]: - [x] Please 🙏🏽 move this code to the main.rb file for OOP it is better to have methods assigned according to the class in this case class app will create the functions for each option and the class main will show the menu of options to select.

File: main.rb [hunk 2]
```diff
@@ -1,21 +1,8 @@
...
+require './app'
 
-require './book'
-require './rental'
-require './student'
-require './person'
-require './classroom'
+def main
+  app = App.new
+  app.run
```
Comment (franclobo) [inline main.rb:6]: - [ ] Please 🙏🏽 add the menu of options here as I mentioned above 👆🏽

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    parent_permission = gets.chomp
+    case parent_permission
+    when 'y' || 'Y'
+      new_student = Student.new(age, name, true)
+      @people << new_student
+      puts 'Person created sucessfully'
+    when 'n' || 'N'
+      new_student = Student.new(age, name, false)
+      @people << new_student
+      puts 'Person created sucessfully'
```
Comment (franclobo) [inline app.rb:50]: - [x] After creating a person student and teacher it didn't return the successful message. Please 🙏🏽 fix this issue for a better UX.

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  def create_teacher
+    print 'Age?: '
+    age = gets.chomp.to_i
+    print 'Name?: '
+    name = gets.chomp
+    print 'Specialization?: '
+    specialization = gets.chomp
+    new_teacher = Teacher.new(age, specialization, name)
+    @people << new_teacher
+    puts 'Person created sucessfully'
```
Comment (franclobo) [inline app.rb:66]: - [x] After creating a person student and teacher it didn't return the successful message. Please 🙏🏽 fix this issue for a better UX.

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    new_book = Book.new(title, author)
+    @books << new_book
+    puts "Book #{title} created succesfully"
```
Comment (franclobo) [inline app.rb:76]: - [x] The same issue as in creating person after creating a book it didn't return the successful message. Please 🙏🏽 fix this issue for a better UX.

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+      book = @books.select { |b| b.title == book_title }
+      puts 'Select the person who is renting the book by the ID'
+      @people.each { |person| puts "Name: #{person.name}, Age: #{person.age}, id: #{person.id}" }
+      person_id = gets.chomp.to_i
+      person = @people.select { |per| per.id == person_id }
+      print 'Please enter a date in the next format yyyy-mm-dd: '
+      date = gets.chomp
+      new_rental = Rental.new(date, book[0], person[0])
+      @rentals << new_rental
+      puts 'Rental created succesfully'
```
Comment (franclobo) [inline app.rb:96]: - [ ] The same issue as in creating person after creating a rental it didn't return the successful message. Please 🙏🏽 fix this issue for a better UX.

File: app.rb [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+  def list_all_people
+    puts 'No people yet, please create a person first' if @people.empty?
+    people.each { |person| puts "Name: #{person.name},  Id: #{person.id}" }
+  end
+
+  def list_all_books
+    puts 'No books yet, please create one' if @books.empty?
+    books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
```
Comment (franclobo) [inline app.rb:126]: - [ ] As the app is not creating the items successfully it is not possible to list them, please 🙏🏽 correct this issue for a better UX.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Alejandro-Bernal-M, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive Pr ✔️ 
- Correct Git flow ✔️ 
- Good UI ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 6 above)
Comment (Alejandro-Bernal-M) [reply app.rb:96]: Good day @franclobo I'm going to correct the menu. I noticed that the message is returning, as you can see even in the screenshot, what do you want me to correct?

File: app.rb (hunk 7 above)
Comment (Alejandro-Bernal-M) [reply app.rb:126]: Hello @franclobo after creating the item you can list them, can you be more specif please about this correction

File: app.rb (hunk 3 above)
Comment (Alejandro-Bernal-M) [reply app.rb:50]: @franclobo  
image

**Review by cynthiainga — APPROVED** [review]
Hi @Alejandro-Bernal-M, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] The app works as expected ✔️ 
- [ ] Descriptive PR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**